### PR TITLE
fix: Kafka headers are slice now

### PR DESCRIPTION
**Pull request by lzap**
Kafka headers are slices instead of map, this is how Kafka does it, we should follow what the library does.

https://issues.redhat.com/browse/HMSPROV-337

### CODE REVIEW

File: internal/kafka/kafka.go [hunk 1]
```diff
@@ -186,9 +186,9 @@ func (b *kafkaBroker) Consume(ctx context.Context, topic string, handler func(ct
 	}
 }
 
-func LookupKafkaHeader(ctx context.Context, headers []protocol.Header) string {
+func Header(name string, headers []protocol.Header) string {
```
Comment (lzap) [inline internal/kafka/kafka.go:189]: Made it generic, also renamed to something shorter if you don't mind.

**Review by adiabramovitch — APPROVED** [review]
LGTM, linter is failing can we fix that?

**Comment by lzap** [comment]
Unexported the function, fixed the check. There is actually a check that detect the variadic function expects even number of arguments and it warns about this. How cool! Disabled for the test which checks for that as well.

**Comment by ezr-ondrej** [comment]
I'm going to be bit pedantic here. The last commit isn't fixing a bug, so it should be `ci:` right? And thus it won't need to be linked to irrelevant jira issue? :)

**Comment by ezr-ondrej** [comment]
I've taken the liberty of fixing the commit message myself so I don't block on it for no reason :)

**Pull request merged**